src/Zed/ruleset.xml

```diff
@@ -100,4 +100,29 @@
         <priority>3</priority>
     </rule>
 
+    <!-- Zed DependencyProvider - RFC Propel Query -->
+    <rule
+        name="ZedDependencyProviderPropelQueryConstantNameRule"
+        message="Zed DependencyProvider - DependencyProvider: {0}"
+        class="ArchitectureSniffer\Zed\DependencyProvider\DependencyProviderPropelQueryConstantNameRule">
+
+        <priority>5</priority>
+    </rule>
+    <rule
+        name="ZedDependencyProviderPropelQueryMethodNameRule"
+        message="Zed DependencyProvider - DependencyProvider: {0}"
+        class="ArchitectureSniffer\Zed\DependencyProvider\DependencyProviderPropelQueryMethodNameRule">
+
+        <priority>5</priority>
+    </rule>
+
+    <!-- Zed Repository - RFC Propel Query -->
+    <rule
+        name="ZedPersistenceRepositoryPropelQueryJoinRule"
+        message="Zed Persistence - Repository: {0}"
+        class="ArchitectureSniffer\Zed\Persistence\Repository\RepositoryPropelQueryJoinRule">
+
+        <priority>5</priority>
```

Comment:
I thought we only have 1, 2, 3, do we need 4 and 5 here?

Reply:
I'll change priority lvl for DependencyProvider and Factory rules. But I think for Repository rule we could let lvl - 5 because it's a complex process.

Reply:
So 4 now, ok.